asciidoc/runtime/src/main/java/io/quarkiverse/qute/web/asciidoc/runtime/AsciidocSectionHelperFactory.java

```diff
@@ -21,11 +29,24 @@ public List<String> getDefaultAliases() {
 
     @Override
     public AsciidocSectionHelper initialize(SectionInitContext context) {
-        return new AsciidocSectionHelper();
+        return new AsciidocSectionHelper(converter);
+    }
+
+    @Override
+    public CompletionStage<Object> resolve(EvalContext context) {
```

Comment:
The disadvantage of this solution (a `ValueResolver` implementation) is that it cannot contribute to validation of type-safe templates, i.e. if there is a type-safe template with `{myString.asciidocify}` then we will not be able to validate this expression and depending on situation the build could fail and something like `quarkus.qute.type-check-excludes=*.asciidocify` would need to be used.

Therefore I recommend to use a regular [extension method](https://quarkus.io/guides/qute-reference#template_extension_methods) instead.

Something like this should work:

```java
@EngineConfiguration
public class AsciidocSectionHelperFactory {

   private static final AsciidocConverter converter = new AsciidocConverter();

    @TemplateExtension(matchNames = {"asciidocify", "toAsciidoc"}) 
    static String convertToAsciidoc(String text) { 
        return converter.apply(text);
    }
}
```

For me `asciidocify` sounds a bit like a tongue-twister. Therefore I recommend to add the `toAsciidoc` alias as well ;-).

And this comment also applies to the MD extension.